### PR TITLE
boost: add version `1.85.0`

### PR DESCRIPTION
**Pull request by gegles**
Specify library name and version:  **boost/1.85.0**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by RubenRBS — DISMISSED** [review]
Thanks a lot!

**Comment by gegles** [comment]
Anybody knows how I can solve this build error:
```
ModuleNotFoundError: No module named 'numpy'
```

Is there a way in the conanfile to specify required Python modules?

**Comment by jwillikers** [comment]
> Anybody knows how I can solve this build error:
> 
> ```
> ModuleNotFoundError: No module named 'numpy'
> ```
> 
> Is there a way in the conanfile to specify required Python modules?

I think that error shows up in current versions of Boost built via Conan, so probably just a red-herring.

**Comment by uilianries** [comment]
I just pushed some changes to make 1.85.0:

- Need to apply the current patch `1.82.0-locale-iconv-library-option` in order to find iconv
- Boost 1.85.0 introduced stacktrace_from_exception (https://www.boost.org/users/history/version_1_85_0.html), and requires disabling this feature in Mac or accepting the risk of leaks.

**Comment by gegles** [comment]
> I just pushed some changes to build 1.85.0:
> 
> * Need to apply the current patch `1.82.0-locale-iconv-library-option` in order to find iconv
> * Boost 1.85.0 introduced stacktrace_from_exception (https://www.boost.org/users/history/version_1_85_0.html), and requires disabling this feature in Mac or accepting the risk of leaks.

Thanks for all your contribs @uilianries! I just did the initial/basic steps, but am stuck with completing this fully. Thanks!

File: recipes/boost/all/dependencies/dependencies-1.85.0.yml [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+  wserialization:
+  - boost_wserialization
+requirements:
+  iostreams:
+  - bzip2
+  - lzma
+  - zlib
+  - zstd
+  locale:
+  - icu
```
Comment (mayeut) [inline recipes/boost/all/dependencies/dependencies-1.85.0.yml:280]:
```suggestion
  - iconv
  - icu
```

iconv was removed when compared to the 1.84.0 file.
with iconv added,  I can cross-build properly for macOS x86_64 (the one failing in cci)

Comment (gegles) [reply recipes/boost/all/dependencies/dependencies-1.85.0.yml:281]: Thanks @mayeut! It indeed seemed to help on macOS... but there still other issues on Linux...

I mostly cared about the `header_only` config so I am not sure how much more I can invest in troubleshooting this... but I am happy to try and incorporate any patches...

**Comment by uilianries** [comment]
@gegles I pushed a new change to make Boost math be built (or not) depending the C++ standard used. 

Since the version 1.82, Boost math uses C++14, but only now (1.85.0) they added a validation int the bjam file for cxx_14 features.

https://github.com/boostorg/math/compare/boost-1.84.0...boost-1.85.0.diff

https://github.com/boostorg/math/blob/boost-1.85.0/README.md#boost-math-library

> ANNOUNCEMENT: This library requires a compliant C++14 compiler.

I was expecting for Boost 1.86, because more libraries deprecated c++03 and will require C++14 in 1.86, but this was a surprise in 1.85

**Comment by gegles** [comment]
> @gegles I pushed a new change to make Boost math be built (or not) depending the C++ standard used.

Thanks @uilianries, much appreciated!

**Comment by gegles** [comment]
Thanks @uilianries for all your hard work on this! Hope this can be merged soon now that it all builds!

**Comment by uilianries** [comment]
> Thanks @uilianries for all your hard work on this! Hope this can be merged soon now that it all builds!

@gegles Thank you for opening and starting it! I was expecting a smooth PR, but Boost Math surprised me. Well, in Boost 1.86.0 we will need to migrate more libs to C++14, but now we have the structure ready for it. Cheers!

**Comment by gegles** [comment]
@uilianries @RubenRBS any ETA on when this can be merged in?

**Review by uilianries — APPROVED** [review]

**Comment by uilianries** [comment]
> @uilianries @RubenRBS any ETA on when this can be merged in?

This PR still needs more reviews.

**Review by jwillikers — APPROVED** [review]

**Pull request merged**